Code/Tools/ProjectManager/Source/ProjectsScreen.cpp

```diff
@@ -408,18 +409,61 @@ namespace O3DE::ProjectManager
             emit ChangeScreenRequest(ProjectManagerScreen::Projects);
         }
     }
+
+    static AZ::IO::FixedMaxPath GetEditorPathForProject(const AZ::IO::FixedMaxPath& projectPath)
+    {
+        AZ::SettingsRegistryImpl settingsRegistry;
+        AZ::IO::FixedMaxPath buildPathSetregPath = projectPath / AZ::SettingsRegistryInterface::DevUserRegistryFolder
+            / "Platform" / AZ_TRAIT_OS_PLATFORM_CODENAME / "build_path.setreg";
+        if (AZ::IO::SystemFile::Exists(buildPathSetregPath.c_str()))
+        {
+            // Merge the build_path.setreg into the local SettingsRegistry instance
+            if (AZ::IO::FixedMaxPath projectBuildPath;
+                settingsRegistry.MergeSettingsFile(buildPathSetregPath.Native(), AZ::SettingsRegistryInterface::Format::JsonMergePatch)
+                && settingsRegistry.Get(projectBuildPath.Native(), AZ::SettingsRegistryMergeUtils::ProjectBuildPath))
+            {
+                // Make a local Settings Registry will be used to merge the build_path.setreg for the supplied projectPath
+                AZ::IO::FixedMaxPath buildConfigurationPath = (projectPath / projectBuildPath).LexicallyNormal();
+
+                // First try <project-build-path>/bin/$<CONFIG> and if that path doesn't exist
+                // try <project-build-path>/bin/$<PLATFORM>/$<CONFIG>
+                buildConfigurationPath /= "bin";
+                if (AZ::IO::FixedMaxPath editorPath = (buildConfigurationPath / AZ_BUILD_CONFIGURATION_TYPE / "Editor").
+                    ReplaceExtension(AZ_TRAIT_OS_EXECUTABLE_EXTENSION);
+                    AZ::IO::SystemFile::Exists(editorPath.c_str()))
+                {
+                    return editorPath;
+                }
+                else if (editorPath = (buildConfigurationPath / AZ_TRAIT_OS_PLATFORM_CODENAME / AZ_BUILD_CONFIGURATION_TYPE / "Editor").
+                    ReplaceExtension(AZ_TRAIT_OS_EXECUTABLE_EXTENSION);
+                    AZ::IO::SystemFile::Exists(editorPath.c_str()))
+                {
+                    return editorPath;
+                }
```

Comment:
For Mac, we'd need to make a platform specific version because the Editor executable would be inside `<project_build_path>/Editor.app/Contents/MacOS/Editor`. But I'm fine with this for now.